client/src/components/Inputs.js

```diff
@@ -0,0 +1,83 @@
+import React, {useEffect, useState, useRef} from 'react';
+import {InputLabel, OutlinedInput, Select, TextField} from '@material-ui/core';
+import PropTypes from 'prop-types';
+
+const handleSelect = (callback) => event => callback(event.target.value);
+
+const OutlinedSelect = (props) => {
+    const inputLabel = useRef(null);
+    const [labelWidth, setLabelWidth] = useState(0);
+
+    useEffect(() => {
+        setLabelWidth(inputLabel.current.offsetWidth)
+    }, []);
+
+    return (
+        <React.Fragment>
+            <InputLabel ref={inputLabel} htmlFor={props.selectId} {...props.inputProps}>{props.labelText}</InputLabel>
+            <Select
+                native
+                value={props.value}
+                onChange={handleSelect(props.setState)}
+                input={
+                    <OutlinedInput
+                        labelWidth={labelWidth}
+                        name={props.selectName}
+                        id={props.selectId}
+                    />
+                }
+                disabled={props.disabled || false}
+                {...props.selectProps}
+            >
+                {props.children}
+            </Select>
+        </React.Fragment>
+    )
+};
+
+OutlinedSelect.propTypes = {
+    children: PropTypes.node.isRequired,
+    selectName: PropTypes.string.isRequired,
+    selectId: PropTypes.string.isRequired,
+    labelText: PropTypes.string,
+    value: PropTypes.string.isRequired,
+    setState: PropTypes.func.isRequired,
+    selectProps: PropTypes.object,
+    inputProps: PropTypes.object,
+    disabled: PropTypes.bool
+};
+
+const CustomOutlinedInput = (props) => {
+    return (
+        <TextField
+            id={props.id || `input-${props.label}`}
+            name={props.name}
+            value={props.value}
+            label={props.label}
+            type={props.type || 'text'}
+            margin="normal"
+            variant="outlined"
+            fullWidth
+            autoComplete={props.autocomplete || 'false'}
+            {...props}
+        />
+    )
+};
+
+CustomOutlinedInput.propTypes = {
+    name: PropTypes.string.isRequired,
+    value: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.object
+    ]).isRequired,
+    label: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+    autocomplete: PropTypes.bool,
+    id: PropTypes.string,
+    type: PropTypes.string
+};
+
+export {
+    OutlinedSelect,
+    CustomOutlinedInput
+}
```

Comment:
a bit nitpicky but I think you can do `export const ComponentName` and you can still destructure it as normal. It can save some lines!

Reply:
Oh ya, slipped my mind for some reason, thanks

Reply:
Actually I just remembered why I am doing this, and I don't know if this actually affects anything other then IDEs but I am defining prop-types outside the component due to the components being functions instead of classes, so I need to put exports after CustomInputOutline.propTypes = {...}